stdlib/socket/src/main/ballerina/src/socket/Module.md

```diff
@@ -1,154 +1,28 @@
-## Module overview
+This module provides an implementation for connecting to a remote socket server or acts as a server for an incoming socket request. The module facilitates three types of endpoints called `socket:Client`, `socket:UdpClient`, and `socket:Listener`.
+#### TCP Client endpoints
+The `socket:Client` is used to connect to a socket server and interact with it. The client can only send the data to the server and the client's `callbackService` can retrieve the data from the server and do multiple requests/responses between the client and the server.
 
-This module provides an implementation for connecting to a remote socket server or acts as a server for an incoming socket request. The module facilitates two types of endpoints called `Client` and `Listener`.
+A Client endpoint can be defined by providing the host and port as follows:
 
-## Samples
-
-### TCP Listener endpoints
-The sample given below shows how a listener is used to listen to the incoming socket request. The `onConnect(socket:Caller)` resource function gets invoked when a new client is connected. The new client is represented using the `socket:Caller`.
-`onReadReady(socket:Caller)` resource gets invoked once the remote client sends some data.
- 
 ```ballerina
-import ballerina/log;
-import ballerina/socket;
-
-listener socket:Listener server = new(61598);
-
-service echoServer on server {
-    resource function onConnect(socket:Caller caller) {
-        log:printInfo("Join: " + caller.remotePort.toString());
-    }
-
-    resource function onReadReady(socket:Caller caller) {
-        var result = caller->read();
-        if (result is [byte[], int]) {
-            var [content, length] = result;
-            if (length > 0) {
-                var writeResult = caller->write(content);
-                if (writeResult is int) {
-                    log:printInfo("Number of bytes written: " 
-                                    + writeResult.toString());
-                } else {
-                    log:printError("Unable to written the content", writeResult);
-                }
-            } else {
-                log:printInfo("Client close: " + caller.remotePort.toString());
-            }
-        } else {
-            log:printError("Unable to read the content", result);
-        }
-    }
-
-    resource function onError(socket:Caller caller, error er) {
-        log:printError("An error occurred", er);
-    }
-}
+socket:Client socketClient = new ({host: "localhost", port: 61598});
```

Comment:
We need to mention the callback service as well here. template of the callback service.

Reply:
And how the client can be used. following lines
```ballerina
    string msg = "Hello Ballerina\n";
    byte[] c1 = msg.toBytes();
    var writeResult = socketClient->write(c1);
```